crates/revm/src/processor.rs

```diff
@@ -294,7 +302,11 @@ impl<'a> BlockExecutor for EVMProcessor<'a> {
         self.stats.apply_post_execution_changes_duration += time.elapsed();
 
         let time = Instant::now();
-        self.db().merge_transitions();
+        let with_reverts = self.tip.map_or(true, |tip| {
+            !self.prune_modes.should_prune_account_history(block.number, tip) &&
+                !self.prune_modes.should_prune_storage_history(block.number, tip)
+        });
+        self.db().merge_transitions(with_reverts);
```

Comment:
makes sense. We collect reverts either if tip is not set, or if pruning tells us to not prune the block we're processing.

Reply:
yup, later on we can address the granularity of the revert retention in the bundle state